### ISSUE TITLE
Windows installer artifacts from Github Actions fail to run

### ISSUE THREAD

**Issue by butlerpd** [SasView/sasview#1981]
So far all three windows Github Actions installers from the release 5.0.5 branch
* SasView-Installer-windows-latest-3.8-d3c75fe3958ad0a22b7f20ee9838e663d99696cf
* SasView-Installer-windows-latest-3.8-dc31ff4dd303f07226aae915c27fa5512dd6867b
* SasView-Installer-windows-latest-3.8-afc53ff4916033e27de1ed04dbddb25049ba8490

And the latest from the main branch
* SasView-Installer-windows-latest-3.8-a2f0665be39cb7d1f5f43605b35c0fd017afc6d2

will install but when they run, the GUI screen loads, but then produces the attached pop up. Clicking close kills SasView while ignoring the pop up and trying to use the GUI anyway produces some responses but nothing that actually allows doing anything.  This happens on both my work laptop (not hooked to VPN) and my cloud windows machine. This is not a problem with the Actions PR installers so I'm not sure where this is coming from, but clearly will need to be fixed before a release if releasing from Actions.

**Comment by butlerpd** [comment]
![SasViewFailNo_guiManager](https://user-images.githubusercontent.com/10955478/143795689-cf6dea2a-8454-4c8c-9a07-b3a1c86faf4c.PNG)

**Comment by krzywon** [comment]
@wpotrzebowski mentioned this same error in slack:

> Hmm, I've been trying to run sasview on OSX from github actions installer and getting the error:
16:28:50 - ERROR: Application failed with: 'Arrow3D' object has no attribute 'do_3d_projection'
16:28:50 - ERROR: Traceback (most recent call last):
 File "sasview.py", line 15, in <module>
 File "sas/qtgui/MainWindow/MainWindow.py", line 119, in run_sasview
AttributeError: 'MainSasViewWindow' object has no attribute 'guiManager'
[13297] Failed to execute script 'sasview' due to unhandled exception: 'MainSasViewWindow' object has no attribute 'guiManager'
Locally it seems to work fine. Is it some dependency issue? Can someone check Windows installer, please: https://github.com/SasView/sasview/releases/tag/untagged-a643ff33301e6fe99d00?

I noticed the same thing, but only after bumps v0.8.1 was released:

> I tested the build I triggered after bumps 0.8.1 was released and am seeing the same error as compared to a build using the same code, pre 0.8.1 release.
Compare my branch build: https://github.com/SasView/sasview/actions/runs/1478142890
to the last main build with the same code from before the v0.8.1 release: https://github.com/SasView/sasview/actions/runs/1455890398

As far as I can tell, `do_3d_projection` is never called directly from `sasview`, `bumps`, `sasmodels`, or `periodictable` and is part of the [matplotlib.Patch3D](https://matplotlib.org/stable/api/_as_gen/mpl_toolkits.mplot3d.art3d.Patch3D.html) class that the [Arrow3D](https://github.com/SasView/sasview/blob/main/src/sas/qtgui/Plotting/Arrow3D.py) class is a child of.

**Comment by krzywon** [comment]
After some digging, I found [this mpl issue](https://github.com/matplotlib/matplotlib/issues/21016) that should be fixed in mpl v3.5.0 but was present in v3.1.x through v3.4.3. I tried running SasView locally with matplotlib v3.4.3 and the GUI loaded as expected. I then installed matplotlib v3.5.0 and the error was thrown when loading SasView from the source. This looks like the change added in mpl v3.5.0 inadvertently broke the Patch3D class.

The builds I linked to in my previous comment were built on Nov. 13, 2021, using mpl v3.4.3, for the build against main, and on Nov. 18, using mpl v3.5.0, for the build I triggered from a branch. mpl v3.5.0 came out on Nov. 16, 2021, and bumps v0.8.1 came out 2 days later, so bumps appears to be a red herring here.

In conclusion - we can't use mpl v3.5.0 and need to pin mpl v3.4.3 or earlier.

**Comment by krzywon** [comment]
Running bumps v0.8.1 and mpl v3.4.3 locally, there are no apparent issues. I dont see any reason to limit bumps. I'm about 95% convinced this is a matplotlib issue

**Comment by wpotrzebowski** [comment]
@krzywon yes it seems like. I've just build these installers: https://github.com/SasView/sasview/releases/tag/untagged-2a10923ac18cb2b09fa7 with `mpl=3.4.3 `and `bumps from master branch ` and at least on OSX seems to work fine.

**Comment by krzywon** [comment]
> @krzywon yes, it seems like. I've just built these installers: https://github.com/SasView/sasview/releases/tag/untagged-2a10923ac18cb2b09fa7 with `mpl=3.4.3 `and `bumps from master branch ` and at least on OSX seems to work fine.

I just tested the Win build. Data loading and fits work fine. Looks like we will need to tag the mpl version for now.
